code/modules/mob/living/blood.dm

```diff
@@ -322,42 +323,39 @@
 		return
 	return /datum/reagent/blood
 
-// This is has more potential uses, and is probably faster than the old proc.
-/proc/get_safe_blood(bloodtype)
-	. = list()
-	if(!bloodtype)
-		return
+/// Returns the blood_type datum that corresponds to the string
+/proc/get_blood_type_by_name(type_name)
+	return GLOB.blood_types[type_name]
 
-	var/static/list/bloodtypes_safe = list(
-		"A-" = list("A-", "O-"),
-		"A+" = list("A-", "A+", "O-", "O+"),
-		"B-" = list("B-", "O-"),
-		"B+" = list("B-", "B+", "O-", "O+"),
-		"AB-" = list("A-", "B-", "O-", "AB-"),
-		"AB+" = list("A-", "A+", "B-", "B+", "O-", "O+", "AB-", "AB+"),
-		"O-" = list("O-"),
-		"O+" = list("O-", "O+"),
-		"L" = list("L"),
-		"U" = list("A-", "A+", "B-", "B+", "O-", "O+", "AB-", "AB+", "L", "U")
-	)
-
-	var/safe = bloodtypes_safe[bloodtype]
-	if(safe)
-		. = safe
+/// Returns the hex color string of a given blood_type datum given an assoc list of blood_DNA e.g. ("Unknown Blood Type", "*X")
+/proc/get_blood_dna_color(list/blood_DNA)
+	var/datum/blood_type/blood_type
+	if(length(blood_DNA))
+		var/last_added_bloodtype_key = blood_DNA[length(blood_DNA)]
+		blood_type = blood_DNA[last_added_bloodtype_key]
```

Comment:
Would be pretty easy to blend all the colors of present blood types together. It's a cool effect (particularly for reducing the prevalence of the "rainbow blood hallway" effect (not the clown blood) so many people dislike)